### ISSUE TITLE
ReactNative v0.75.4 with new architecture android app is crashing 

### ISSUE THREAD

**Issue by sankar-gp** [react-navigation/react-navigation#12183]
### Current behavior

ReactNative v0.75.4 with new architecture android app is crashing when I click back. But it's working in iOS and when I disable new arch it's working fine. Issue is only with newArchEnabled=true

Error Log:
Cannot remove child at index 1 from parent ViewGroup[1332], only 2 children in parent. Warning: childCount may be incorrect! null

### Expected behavior

App should not crash when newArchEnabled=true

### Reproduction

norepo

### Platform

- [X] Android
- [ ] iOS
- [ ] Web
- [ ] Windows
- [X] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [X] @react-navigation/stack
- [ ] @react-navigation/native-stack
- [ ] react-native-tab-view

### Environment

- [] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | ^6.1.18
| @react-navigation/bottom-tabs          |^6.6.1
| @react-navigation/material-top-tabs    |^6.6.14
| @react-navigation/native-stack         |^6.11.0
| react-native-screens                   | ^3.34.0
| react-native-gesture-handler           |^2.20.0
| react-native-tab-view                  |^3.5.2
| react-native-pager-view                |^6.4.1
| react-native                           |0.75.4
| node                                   |>=18
| npm or yarn                            |yarn@3.6.4

**Comment by sankar-gp** [comment]
> Couldn't find version numbers for the following packages in the issue:
> 
> * `@react-navigation/stack`
> 
> Can you update the issue to include version numbers for those packages? The version numbers must match the format 1.2.3.


@react-navigation/native-stack | 6.11.0

**Comment by spyshower** [comment]
@sankar-gp I have been encountering this problem for quite some time too. It's probably related to flatlists.

**Comment by sankar-gp** [comment]
@spyshower  Do you have any fix?

**Comment by spyshower** [comment]
@sankar-gp Nope, I am still in development so I switched to scrollviews for now. I think they are working on a fix on the react-native-screens repo

**Comment by WKampel** [comment]
Same issue

**Comment by spyshower** [comment]
I think it was supposed to be fixed in `react-native-screens` beta 9 but I still encounter the issue on that version :(

**Comment by shovel-kun** [comment]
@spyshower They pushed a hotfix. Does this fix the issue?

https://github.com/software-mansion/react-native-screens/pull/2403

**Comment by spyshower** [comment]
@shovel-kun yes, it appears that it does!

**Comment by naufalotomedia** [comment]
any update on this ?

**Comment by shovel-kun** [comment]
@naufalotomedia Use the latest react-native-screens version (`@4.0.0-beta.16`)

**Comment by naufalotomedia** [comment]
> [@naufalotomedia](https://github.com/naufalotomedia) Use the latest react-native-screens version (`@4.0.0-beta.16`)

this is legit and work perfectly, thank you in advance

**Comment by migueldaipre** [comment]
As @shovel-kun mentioned,

It is fixed in the latest version of react-native-screens. (@4.0.0-beta.16)

Closing.
